x-pack/plugins/infra/server/lib/alerting/inventory_metric_threshold/register_inventory_metric_threshold_rule_type.ts

```diff
@@ -102,6 +110,9 @@ export async function registerMetricInventoryThresholdRuleType(
       context: [
         { name: 'group', description: groupActionVariableDescription },
         { name: 'alertState', description: alertStateActionVariableDescription },
+        getAlertDetailsPageEnabledForApp(config, 'metrics')
+          ? { name: 'alertDetailsUrl', description: alertDetailUrlActionVariableDescription }
+          : null,
```

Comment:
Nit: You could avoid the `.filter()` step by conditionally merging in an empty array:

```typescript
        ...(getAlertDetailsPageEnabledForApp(config, 'metrics')
          ? [{ name: 'alertDetailsUrl', description: alertDetailUrlActionVariableDescription }]
          : []),
```

Reply:
@weltenwort Also a good solution. Will update.